lib/Driver/Driver.cpp

```diff
@@ -1749,6 +1749,16 @@ void Driver::buildActions(SmallVectorImpl<const Action *> &TopLevelActions,
   switch (OI.CompilerMode) {
   case OutputInfo::Mode::StandardCompile: {
 
+    // If we're not compiling the standard library directly, add an action
+    // before any others that will quickly load the standard library module.
+    // This will ensure that, if we need to build the standard library from
+    // a module interface, it happens once, rather than once per parallel
+    // invocation.
+    if (!Args.hasArg(options::OPT_parse_stdlib)) {
+      auto loadModule = C.createAction<LoadModuleJobAction>(STDLIB_NAME);
+      TopLevelActions.push_back(loadModule);
+    }
```

Comment:
If nothing else depends on it, it won't necessarily happen first. (Check out the tests using `-driver-print-bindings` for testing dependency info.)